tls/s2n_ktls_io.c

```diff
@@ -271,3 +271,55 @@ S2N_RESULT s2n_ktls_recvmsg(struct s2n_connection *conn, uint8_t *record_type, u
     *bytes_read = result;
     return S2N_RESULT_OK;
 }
+
+static S2N_RESULT s2n_ktls_new_iovecs_with_offset(const struct iovec *bufs,
```

Comment:
This impl is missing the case where the offset is aligned with a iovec boundary. In that case we don't need to modify/copy anything.

It also seems a bit wasteful to allocate 1000 iovecs only to use the last one because the offset points to that. Couldn't we lazily realloc `mem` when we hit the index that contains the offset?

Reply:
I was trying to keep the initial version simple and planned to optimize later. Like, we also discussed using the heap for small numbers of iovecs.

But I can do some of the optimizations now, if you don't think it'll complicate this PR.

Reply:
I think it's fine for now. Do you want to open an issue to track it is or is it a fast follow?